components/display/max7219digit.rst

```diff
@@ -0,0 +1,181 @@
+commands have been added to the basic display set.
+
+.. code-block:: yaml
+
+    display:
+      - platform: max7219digit
+        cs_pin: D8
+        num_chips: 4
+        lambda: |-
+          it.strftime(0, 0, id(digit_font), "%H:%M", id(hass_time).now());
+          it.image(24, 0, id(my_image));
+          it.line(1, 7, 21, 7);
+    font:
+      - file: "pixelmix.ttf"
+        id: digit_font
+        size: 6
+
+    time:
+      - platform: homeassistant
+        id: hass_time
+
+    image:
+      - file: "smile.png"
+        id: my_image
+
+This is roughly the code used to display the MAX7219 pictured in the image.
+
+Some special MAX7219 digit code can be added as follows:
+
+.. code-block:: yaml
+
+    display:
+      - platform: max7219digit
+        # ...
+        lambda: |-
+          // Print 0 at position 0 (left)
+          it.print(0,0, id(digit_font), "Hello!");
+          it.scroll(true,0,100,5000,1500);
+
+By default the MAX7219Digit display has scroll enabled. The paramaters can be set in the YAML file.
+They can also be changed in the Lambda by adding the following command:
+it.scroll(ON/OFF,MODE,SPEED,DELAY,DWELL).
+
+- ON/OFF -> switch scrolling on or off
+- MODE -> 0 = Contineous scrolling
+-         1 = Stop at end and reset
+- SPEED -> Set speed of scrolling (ms for every step of one dot)
+- DELAY -> pauze at start of scrolling
+- DWELL -> pauze at end of scrolling (only in mode 1)
+
+.. code-block:: yaml
+
+    display:
+      - platform: max7219digit
+        # ...
+        lambda: |-
+          # ...
+          it.scroll(true,0,100,5000,1500);
+          // OR
+          it.scroll(true,0,);
+          // OR
+          it.scroll(true);
+
+- The screen does not scroll if the text fits within the screen.
+- Printdigit("XXXXXXXXX") and printfdigit("XXXXXX") the alternative way of displaying text does not scroll
+
+.. code-block:: yaml
+
+    display:
+      - platform: max7219digit
+        # ...
+        lambda: |-
+          it.invert_on_off(true);
+          // Print Hello at position 0 (left)
+          it.print(0,0, id(digit_font), "Hello!");
+
+The function it.invert_on_off(true); will invert the display. So background pixels are on and texts pixels are
+off. it.invert_on_off(false); sets the display back to normal. In case no argument is used: it.inverst_on_off();
+the inversion will toggle from on to off or visa versa. This will happen every time the display is updated.
+So a blinking effect is created. The background pixels are only set at the next update, the pixels drawn in
+the various function like print, line, etc. are directly influenced by the invert command.
+
+.. code-block:: yaml
+
+    display:
+      - platform: max7219digit
+        # ...
+        lambda: |-
+          // Print Hello at position 0 (left)
+          it.print(0,0, id(digit_font), "Hello!");
+          it.invert_on_off(true);
+          it.line(0,0,32,8);
```

Comment:
This seems going off buffer

Also example said it will clear the pixels. But when I run it, it inverted the pixels. 

Also line functions and other graphics accept a 5th parameter which be default is COLOR_ON. But you can pass color off so you actually clear pixels this way.

Reply:
code has been changed, the output of the example is quite ugly but works.